rust/src/dataset.rs

```diff
@@ -184,6 +178,32 @@ impl Dataset {
             ));
         }
 
+        // Running checks for the different write modes
+        // create + dataset already exists = error
+        if object_store.exists(&latest_manifest_path).await?
+            && matches!(params.mode, WriteMode::Create)
+        {
+            return Err(Error::IO(format!("Dataset already exists: {uri}")));
+        }
+
+        // append + dataset doesn't already exists = warn + switch to create mode
+        if !object_store.exists(&latest_manifest_path).await?
```

Comment:
this only needs to be checked once and just reuse the boolean result

Reply:
Done. Might have incurred extra network call.